saleor/graphql/channel/mutations/channel_update.py

```diff
@@ -218,9 +222,26 @@ def _update_warehouses(cls, instance, cleaned_data):
         if remove_warehouses:
             instance.warehouses.remove(*remove_warehouses)
 
+    @classmethod
+    def _update_voucher_usage(cls, cleaned_input, instance):
+        """Update voucher code usage.
+
+        When the 'include_draft_order_in_voucher_usage' flag is changed:
+        - True -> False: decrease voucher usage of all vouchers associated with
+        draft orders.
+        - False -> True: disconnect vouchers from all draft orders.
+        """
+        current_flag = cleaned_input.get("include_draft_order_in_voucher_usage")
+        previous_flag = cleaned_input.get("prev_include_draft_order_in_voucher_usage")
+        if current_flag is False and previous_flag is True:
+            decrease_voucher_code_usage_of_draft_orders(instance.id)
+        elif current_flag is True and previous_flag is False:
+            disconnect_voucher_codes_from_draft_orders(instance.id)
```

Comment:
Are we sure about this? 
When I disable `include_draft_order_in_voucher_usage`, I would expect the voucher.usage_limit will not increase when using it with draft order. But currently, it fully removes the vouchers from the draft orders, and it sounds to me like a separate setting or enum setting.

Reply:
Nevermind. I understand the flow now.

Reply:
When you disabling this setting, the voucher usage is decreased, as draft order's vouchers are not counted into voucher usage anymore.

When switching on this setting, all vouchers are removed from draft orders, as it to many cases to handle. For example when the voucher will have `usage_limit=2` and 3 draft orders will have this voucher assigned there is no easy wat to handle such case.